### DISCUSSION TITLE
Gutenberg support

### DISCUSSION THREAD

**Discussion by ranostaj** [flyntwp/flynt#535] [General]
Hey there, I am considering implementing Flynt for one of my clients due to its components-based architecture. However, the client is accustomed to using Gutenberg for content creation. I would like to maintain this editing experience for them. Could you please advise on how to seamlessly integrate Flynt components into the Gutenberg editor as blocks, is there any way? This feature would greatly enhance the flexibility in structuring pages. Thank you.

**Comment by aaronmeder** [comment]
Hey @ranostaj 

I am not officially part of the Flynt team, but as far as I know there is no way to directly and easily integrate Flynt components into Gutenberg. Flynt since v2 uses Gutenberg for editing posts - but uses Flynt components for the rest of the site and pages (so no Full Site Editing experience). 

If a customer is looking for the visual way that Gutenberg provides (for more than posts) then I'd recommend to embrace that fully. You can keep the effort to create custom blocks simpler by using `[ACF Blocks](https://www.advancedcustomfields.com/resources/blocks/)`.

Aaron

**Comment by timohubois** [comment]
@ranostaj and @aaronmeder a few days ago I was interested in, if it easily possible to use exiting Components with Gutenberg and i created a [branch](https://github.com/flyntwp/flynt/tree/Timo/UseComponentsAsBlock) for testing purposes.

In general there is one new file (`blocks.php`) in the inc folder and some minor adjustments of styles. To register a Block you can create a `block.json` (with [ACF Schema](https://advancedcustomfields.com/schemas/json/main/block.json)) and this should it be. These components should then work as usual in the classic editor and accordingly in Gutenberg.

The full changes compared to the current master branch can be found here: https://github.com/flyntwp/flynt/compare/master...Timo/UseComponentsAsBlock

**Notice: This approach is not officially supported or part of Flynt at the moment.** I also noticed some downsides and issues with the Gutenberg UI and did not dive into details then, due I was primarily interested in finding out whether existing components in Gutenberg could be used with little effort.

However, I think as a starting point it may can help ✌️

**Comment by Flowgram** [comment]
Hey @timohubois, thanks for looking into this. 

Are there any plans to add full Gutenberg support for ACF blocks in the future? As much as I prefer the ACF flexible content field setup as a page builder, I think we need to be aligning ourselves with the core WordPress experience as this is what some clients expect. Ideally we'd support both options.

Loving Flynt btw!
